File: app/src/main/java/com/evaluation/fragment/DetailFragment.java
package com.evaluation.fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;

import com.bumptech.glide.Glide;
import com.evaluation.userlist.MainActivity;
import com.evaluation.userlist.R;
import com.evaluation.utils.StringUtils;
import com.evaluation.viewmodel.PageViewModel;

import org.jetbrains.annotations.NotNull;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * @author Vladyslav Havrylenko
 * @since 09.03.2020
 */
public class DetailFragment extends Fragment {

    public final String TAG = DetailFragment.class.getCanonicalName();

    private PageViewModel mPageViewModel;

    private MainActivity mActivity;

    private View mRootView;

    @BindView(R.id.picture)
    ImageView mPictureView;

    @BindView(R.id.gender)
    TextView mGenderView;

    @BindView(R.id.name)
    TextView mNameView;

    @BindView(R.id.location)
    TextView mLocationView;

    @BindView(R.id.email)
    TextView mEmailView;

    @BindView(R.id.phone)
    TextView mPhoneView;

    @BindView(R.id.dob)
    TextView mDobView;

    @BindView(R.id.nat)
    TextView mNatView;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        mPageViewModel = new ViewModelProvider(requireActivity()).get(PageViewModel.class);
    }

    @Nullable
    @Override
    public View onCreateView(@NotNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        if (mRootView == null) {
            mRootView = inflater.inflate(R.layout.info_layout, container, false);
            ButterKnife.bind(this, mRootView);
            loadInfo();
        }
        return mRootView;
    }

    @Override
    public void onActivityCreated(Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        mActivity = (MainActivity) getActivity();
    }

    private void loadInfo() {
        mPageViewModel.getResult().observe(requireActivity(), result -> {
            Glide.with(mActivity)
                    .load(result.getPicture().getLarge())
                    .into(mPictureView);

            String firstLastName = result.getName().getTitle() +
                    StringUtils.DOT_STRING +
                    StringUtils.BLANK_STRING +
                    result.getName().getFirst() +
                    StringUtils.BLANK_STRING +
                    result.getName().getLast();

            mGenderView.setText(result.getGender().toUpperCase());
            mNameView.setText(firstLastName);

            String address = result.getLocation().getCountry() +
                    StringUtils.COMMA_STRING +
                    StringUtils.BLANK_STRING +
                    result.getLocation().getCity() +
                    StringUtils.COMMA_STRING +
                    StringUtils.BLANK_STRING +
                    result.getLocation().getStreet().getName() +
                    StringUtils.BLANK_STRING +
                    result.getLocation().getStreet().getNumber();

            mLocationView.setText(address);
            mEmailView.setText(result.getEmail());
            mPhoneView.setText(result.getPhone());

            String birthday = result.getDob().getDate() +
                    StringUtils.COMMA_STRING +
                    StringUtils.BLANK_STRING +
                    result.getDob().getAge();

            mDobView.setText(birthday);
            mNatView.setText(result.getNat());

        });
    }
}

File: app/src/main/java/com/evaluation/fragment/DetailFragment.kt
package com.evaluation.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import butterknife.BindView
import butterknife.ButterKnife
import com.bumptech.glide.Glide
import com.evaluation.model.result.Result
import com.evaluation.userlist.MainActivity
import com.evaluation.userlist.R
import com.evaluation.utils.StringUtils
import com.evaluation.viewmodel.PageViewModel

/**
 * @author Vladyslav Havrylenko
 * @since 09.03.2020
 */
class DetailFragment : Fragment() {

    val TAG = DetailFragment::class.java.canonicalName

    lateinit var mPageViewModel: PageViewModel

    lateinit var mActivity: MainActivity

    private var mRootView: View? = null

    @BindView(R.id.picture)
    lateinit var mPictureView: ImageView

    @BindView(R.id.gender)
    lateinit var mGenderView: TextView

    @BindView(R.id.name)
    lateinit var mNameView: TextView

    @BindView(R.id.location)
    lateinit var mLocationView: TextView

    @BindView(R.id.email)
    lateinit var mEmailView: TextView

    @BindView(R.id.phone)
    lateinit var mPhoneView: TextView

    @BindView(R.id.dob)
    lateinit var mDobView: TextView

    @BindView(R.id.nat)
    lateinit var mNatView: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mPageViewModel = ViewModelProvider(requireActivity()).get(PageViewModel::class.java)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        if (mRootView == null) {
            val view = inflater.inflate(R.layout.info_layout, container, false)
            ButterKnife.bind(this, view)
            mRootView = view
            loadInfo()
        }
        return mRootView
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        mActivity = activity as MainActivity
    }

    private fun loadInfo() {
        mPageViewModel.result.observe(
            requireActivity(),
            Observer { result: Result ->

                activity?.let {
                    Glide.with(it)
                        .load(result.mPicture.mLarge)
                        .into(mPictureView)
                }

                val firstLastName = result.mName.mTitle +
                        StringUtils.DOT_STRING +
                        StringUtils.BLANK_STRING +
                        result.mName.mFirst +
                        StringUtils.BLANK_STRING +
                        result.mName.mLast

                mGenderView.text = result.mGender.toUpperCase()
                mNameView.text = firstLastName

                val address = result.mLocation.mCountry +
                        StringUtils.COMMA_STRING +
                        StringUtils.BLANK_STRING +
                        result.mLocation.mCity +
                        StringUtils.COMMA_STRING +
                        StringUtils.BLANK_STRING +
                        result.mLocation.mStreet.mName +
                        StringUtils.BLANK_STRING +
                        result.mLocation.mStreet.mNumber

                mLocationView.text = address
                mEmailView.text = result.mEmail
                mPhoneView.text = result.mPhone

                val birthday = result.mDob.mDate +
                        StringUtils.COMMA_STRING +
                        StringUtils.BLANK_STRING +
                        result.mDob.mAge

                mDobView.text = birthday
                mNatView.text = result.mNat
            }
        )
    }
}